[jerremyng] iP

jerremyng opened the pull request:
# FakeGPT

> Somebody once told me the world is gonna roll me - Shrek

`FakeGPT` is a knockoff version of gpt, that acts as a task manager for all your needs.
- Easy to use!
- Very good
- 10/10 rated
- ~~sentient~~

Set up
1. Download [here](https://github.com/jerremyng/ip/releases)
2. Add tasks
3. **Be free**

Features:
- [x] Remembers your work!
- [ ] Reminds you of deadlines (WIP)

`main` method:
````java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
````

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

jerremyng committed:
Code for hi and bye!

jerremyng committed:
Echos command

jerremyng committed:
Added a taskList class that serves as memory. It has a add and toString method.

jerremyng committed:
Added marking as done functionality. Also a task class, to encapsulate task state + behaviour

jerremyng committed:
Added big changes, namely 3 subtasks that inherit from Task class. Also added logic to parse inputs. TODO: add some error handling

jerremyng committed:
Added test case for general use (wrong input not checked/implemented yet)

jerremyng committed:
Many changes, created a parser class as well as command class. There is now a hierachy of command types inheriting from command. Also added error handling for most cases I could detect. Need to work on test cases!

jerremyng committed:
Added many changes, delete function and fixed up testing script to test for common use cases and a little typos!

jerremyng committed:
Add storage class to store data through sessions

jerremyng committed:
Improve comments

jerremyng committed:
Add new storage feature for duke
Merge branch 'branch-Level-7'

jerremyng committed:
Store dates in deadline tasks as LocalDate objects.

A very basic implementation that only takes in "dd MMM yyyy" format. Note MMM must be capitalized ie Jan, Feb instead of jan, feb.

jerremyng committed:
Use LocalDate objects in Event tasks.

jerremyng committed:
Merge branch 'branch-Level-8'.

This makes Tasks in the chatbot use LocalDate objects, for better control over data.
Merge branch 'branch-Level-8'

jerremyng committed:
Refactor code to add a new UI class in charge of all I/O operations.

jerremyng committed:
Merge branch 'branch-A-MoreOOP'

This merge refactors code to improve OOP applications.

jerremyng committed:
Fix some naming to follow conventions.

jerremyng committed:
Create task package.
Refactors Task related classes into a package.

jerremyng committed:
Refactor commands and exceptions into packages.

jerremyng committed:
Add factory method to Task, so that classes outside only use factory to create tasks.

jerremyng committed:
Merge branch 'branch-A-Packages'

Accept package refactor and factory method in task creation.

jerremyng committed:
Merge branch 'add-gradle-support'

jerremyng committed:
Add test case for Parser class. Covers most use cases.

jerremyng committed:
Add test case for TaskList class.

jerremyng committed:
Add javadocs to all files. JavaDocs follow all conventions.

jerremyng committed:
Fix coding standard violations

Import statements use * imports.

Using * imports makes it difficult for other developers as the classes used are not explitcity defined.

Let's
* Update imports to be explicit.
* Fix any coding standard violations.

jerremyng committed:
Add find command

There is no way for users to search through their tasks.

A find command will improve user experience allowing them to search for tasks.

Lets add a find command.

jerremyng committed:
Merge branch 'branch-A-JavaDoc'

Adds javadocs.

jerremyng committed:
Merge branch 'branch-Level-9'

jerremyng committed:
Resolve merge conflict

ThitipatC commented on src/main/java/duke/task/TaskList.java:
```diff
@@ -0,0 +1,102 @@
+ */
+public class TaskList implements Iterable<Task>{
+    /** Stores tasks in a list */
+    private List<Task> taskList = new ArrayList<>();
```
listOfTasks or tasksList (plural names) to indicate that it is a collection of task might be clearer.

ThitipatC reviewed (commented):
LKGTM. I spent good amount of time looking at it and only found one possible minor naming convention.

jerremyng committed:
Fix checkstyle violations to follow coding standards

jerremyng committed:
Merge branch 'branch-A-CheckStyle'

jerremyng committed:
Add GUI to duke

jerremyng committed:
Fix checkstyle violations

jerremyng committed:
Merge branch 'branch-Level-10'

Tsenrae commented on src/main/java/duke/command/Command.java:
```diff
@@ -0,0 +1,38 @@
+        return this.response;
+    }
+
+    public void setResponse(String response) {
```
javaDoc missing for method setResponse()

Tsenrae commented on src/main/java/duke/exceptions/MissingInfoException.java:
good addition of other exceptions

Tsenrae reviewed (commented):
LGTM! Nice usage of OOP abstraction, especially the comands and exceptions

jerremyng committed:
Fix code quality

There are several code quality violations and possible refactors in this codebase.

Correcting these to follow the standard would increase code quality.

Let's fix these to follow the given standards.

jerremyng committed:
Implement assertions

Assertions are not used in this code.

Assertions are powerful tools to catch mistake. Using them will improve robustness of the code.

Let's implement some of them.

jerremyng committed:
Merge pull request #1 from jerremyng/branch-A-CodeQuality

Fix code quality

jerremyng committed:
Merge pull request #2 from jerremyng/branch-A-Assertions

Implement assertions

jerremyng committed:
Add function to sort list based on dates.

Items are ordered by order of insertion, finding the next task to do may be difficult.

A sort command will help users prioritize tasks better.

Lets add a sort functionality.

jerremyng committed:
Merge branch 'branch-C-Sort'

The branch C-Sort implements sorting of the task list.

This feature can help users find tasks chronologically and improves user experience.

Lets merge this branch.

jerremyng committed:
Add user guide

There is no way for new users to be able to quickly learn this program.

Adding a user guide will help improve onboarding process.

Lets add a user guide.

jerremyng committed:
Fix test cases

Some test cases need updating for changes.

Lets update them.

jerremyng committed:
Polish and clean up codebase

Some best practices are not implemented.

Lets correct them.
